x-pack/legacy/plugins/lens/public/indexpattern_datasource/dimension_panel/dimension_panel.tsx

```diff
@@ -41,152 +50,185 @@ export interface OperationFieldSupportMatrix {
+    // TODO: What do we do if we couldn't find a column?
+    return false;
+  }
+
+  const operationsForNewField =
+    operationFieldSupportMatrix.operationByField[droppedItem.field.name];
+
+  const layerId = props.layerId;
+  const selectedColumn: IndexPatternColumn | null =
+    props.state.layers[layerId].columns[props.columnId] || null;
+  const currentIndexPattern =
+    props.state.indexPatterns[props.state.layers[layerId]?.indexPatternId];
+
+  // We need to check if dragging in a new field, was just a field change on the same
+  // index pattern and on the same operations (therefore checking if the new field supports
+  // our previous operation)
+  const hasFieldChanged =
+    selectedColumn &&
+    hasField(selectedColumn) &&
+    selectedColumn.sourceField !== droppedItem.field.name &&
+    operationsForNewField &&
+    operationsForNewField.includes(selectedColumn.operationType);
+
+  // If only the field has changed use the onFieldChange method on the operation to get the
+  // new column, otherwise use the regular buildColumn to get a new column.
+  const newColumn = hasFieldChanged
+    ? changeField(selectedColumn, currentIndexPattern, droppedItem.field)
+    : buildColumn({
+        op: operationsForNewField ? operationsForNewField[0] : undefined,
+        columns: props.state.layers[props.layerId].columns,
+        indexPattern: currentIndexPattern,
+        layerId,
+        suggestedPriority: props.suggestedPriority,
+        field: droppedItem.field,
+        previousColumn: selectedColumn,
+      });
+
+  trackUiEvent('drop_onto_dimension');
+  const hasData = Object.values(props.state.layers).some(({ columns }) => columns.length);
+  trackUiEvent(hasData ? 'drop_non_empty' : 'drop_empty');
+
+  props.setState(
+    changeColumn({
+      state: props.state,
+      layerId,
+      columnId: props.columnId,
+      newColumn,
+      // If the field has changed, the onFieldChange method needs to take care of everything including moving
+      // over params. If we create a new column above we want changeColumn to move over params.
+      keepParams: !hasFieldChanged,
+    })
+  );
+
+  return true;
+}
+
+export const IndexPatternDimensionTriggerComponent = function IndexPatternDimensionTrigger(
+  props: IndexPatternDimensionTriggerProps
+) {
+  const layerId = props.layerId;
+
+  const selectedColumn: IndexPatternColumn | null =
+    props.state.layers[layerId].columns[props.columnId] || null;
 
-    return (
-      isDraggedField(dragging) &&
-      layerIndexPatternId === dragging.indexPatternId &&
-      Boolean(hasOperationForField(dragging.field))
-    );
+  const { columnId, uniqueLabel } = props;
+  if (!selectedColumn) {
+    return null;
   }
+  return (
+    <EuiLink
+      id={columnId}
+      className="lnsConfigPanel__triggerLink"
+      onClick={() => {
+        props.togglePopover();
+      }}
+      data-test-subj="lns-dimensionTrigger"
+      aria-label={i18n.translate('xpack.lens.configure.editConfig', {
```

Comment:
We're trying not to have two `i18n.translate` calls with the same id in it, since it's prone to introduce errors where default messages vary later on. Could we please just call this once and store it within a variable that is used here.

Reply:
It's not possible to have two i18n IDs with different default messages. It's checked automatically. How important is this?